dipy/workflows/align.py

```diff
@@ -45,16 +54,464 @@ def run(self, input_files, new_vox_size, order=1, mode='constant', cval=0,
+            ault '32')
+
+        sampling_prop : int, optional
+            Number ([0-100]) of voxels to be used for calculati
+            ng the PDF. 'None' implies all voxels. (default 'None')
+
+        metric : string, optional
+            The similarity metric to be used for gathering mutual information
+            . (default 'Mutual Information metric')
+
+        level_iters : variable int, optional
+            the number of iterations at each scale of the scale space.
+            `level_iters[0]` corresponds to the coarsest scale,
+            `level_iters[-1]` the finest, where n is the length of the
+            sequence. By default, a 3-level scale space with iterations
+            sequence equal to [10000, 1000, 100] will be used.
+
+        sigmas : variable floats, optional
+            custom smoothing parameter to build the scale space (one parameter
+            for each scale). By default, the sequence of sigmas will be
+            [3, 1, 0].
+
+        factors : variable floats, optional
+            custom scale factors to build the scale space (one factor for each
+            scale). By default, the sequence of factors will be [4, 2, 1].
+
+        progressive : boolean, optional
+            Flag for enabling/disabling the progressive registration.
+            (default 'True')
+
+
+        save_metric : boolean, optional
+            If true, the metric values are
+            saved in a file called 'quality_metric.txt'
+            (default 'False')
+
+            By default, the similarity measure
+            values such as the distance and the
+            metric of optimal parameters is only
+            displayed but not saved.
+
+        out_dir : string, optional
+            Directory to save the transformed image and the affine matrix.
+            (default '')
+
+        out_moved : string, optional
+            Name for the saved transformed image.
+            (default 'moved.nii.gz')
+
+        out_affine : string, optional
+            Name for the saved affine matrix.
+            (default 'affine.txt')
+
+        out_quality : string, optional
+            Name of the file containing the saved quality
+            metric (default 'quality_metric.txt')
+
+        """
+
+        """
+        setting up the io iterator to gobble the input and output paths
+        """
+        io_it = self.get_io_iterator()
+
+        for static_img, mov_img, moved_file, affine_matrix_file, \
+                qual_val_file in io_it:
+
+            """
+            Load the data from the input files and store into objects.
+            """
+            image = nib.load(static_img)
+            static = np.array(image.get_data())
+            static_grid2world = image.affine
+
+            image = nib.load(mov_img)
+            moving = np.array(image.get_data())
+            moving_grid2world = image.affine
+
+            self.check_dimensions(static, moving)
+
+            if transform.lower() == 'com':
```

Comment:
Instead of writing `transform.lower()` 4 times, you could put it in a variable.

Reply:
@nilgoyette: Yes, you are right. Thanks for this one. Much appreciate it. Will update the code.

Reply:
Done.